### ISSUE TITLE
Form.Autosuggest Accessibility

### ISSUE THREAD

**Issue by MaxFrank13** [openedx/paragon#2410]
# Improvements

There are a number of adjustments that can be made to the `Form.Autosuggest` component to make it more accessible. Some of these adjustments are relatively trivial and involve simply updating the attributes on a few of the HTML elements. However, other changes will require a bit of refactoring around the HTML structure, especially with regards to how labels are associated with the `<input>` element.
________
## Resources

[Accessible Autocomplete examples](https://alphagov.github.io/accessible-autocomplete/examples/)
[Anatomy of an Accessible Auto Suggest](https://uxmastery.com/anatomy-of-an-accessible-auto-suggest/)
[Best practices for the accessibility of an autocompletion component](https://a11y-guidelines.orange.com/en/articles/autocomplete-component/)
______
## Moving forward

Aperture team has plans to address these issues over time, but the work has not yet been assigned so we welcome any assistance others can provide. If any updates are made regarding the accessibility of the `Form.Autosuggest` component, please reference this issue and don't hesitate to reach out to  **@openedx/2u-aperture**.

```[tasklist]
### Tasks
- [ ] https://github.com/openedx/paragon/issues/2435
- [ ] https://github.com/openedx/paragon/issues/2436
- [ ] https://github.com/openedx/paragon/issues/2437
- [ ] https://github.com/openedx/paragon/issues/2438
```

**Comment by brian-smith-tcril** [comment]
Great writeup here! I'm thinking it might make sense to split this up into a couple issues:
* Trivial fixes/more complex attributes
* Label Association (this seems like it could use some discovery/discussion around how we want to handle it, I think it'd be nice for those conversations to live on their own issue)
* Tab Behavior (this also could use some discussion, I'm not sure what best practices are for tabbing through/out of a suggestions dropdown)

**Comment by wittjeff** [comment]
Regarding focus behavior, I refer to the linked articles outlining or demonstrating accessible autosuggest requirements.  Additional background re aria-activedescendant: this attribute is intended to be a kind of "sub-focus". So a parent item has focus, but you're really working on some kind of child item.  The two examples that come to mind are grids (parent has focus, active cell has aria-activedescendant) and comboboxes (input has focus, menu items have activedescendant.). 

One thing to know here is that there are sometimes two ways to implement theses common patterns; you can make a grid where the cell being modified has focus, or where some control within that cell has focus.

  Aria-activedescendant is intended to guide a screen magnifier so it can keep the thing you're working on in the magnification frame at all times.  I'm not entirely sure of the level of support across accessibility tools.

Regarding aria-live: be aware that testing with VoiceOver on Mac OS with Chrome may not work as expected, particularly with aria-live. Instead use VoiceOver with Safari.  This is a longstanding issue; I don't think Apple feels strongly motivated to fix Chrome compatibility bugs.

**Comment by brian-smith-tcril** [comment]
Should be fixed by now. If not can reopen.
